### ISSUE TITLE
Support for DICOM

### ISSUE THREAD

**Issue by sarthakpati** [bayer-science-for-a-better-life/tiffslide#32]
Hey,

It would be great to add the ability to support WSI DICOM. There is a very nice library called [wsidicom](https://github.com/imi-bigpicture/wsidicom) that takes care of it in a nice way. 

Cheers,
Sarthak

**Comment by sarthakpati** [comment]
Yup, it is the same on our side. I'll put this on our list, too. 😄Will coordinate here.

**Comment by benmalef** [comment]
Hey, 
I hope you are doing well.
We have had a little discussion with @sarthakpati about WSI DICOM support.  
I  would like to contribute to this feature. 😄
Any guidance or help would be greatly appreciated

Cheers,
Veniamin

**Comment by ap--** [comment]
Hi @benmalef and @sarthakpati 

Thank you for offering to contribute!
It's been a long time since I've looked into DICOM support.

The basic idea for supporting file formats that can not be read by tifffile, is to provide an Array like interface to the new format, which then lazily loads only the data needed to return a specific slice.
If the reader that provides this interface can work on file-like objects you get cloud support for free.

You can look at the open draft PR for mirax support, where I use kerchunk to build an index for reading the compressed image data via zarr. You could do something similar for DICOM I suppose. Another option is to look at something like pydicom, and see if it either provides that array interface, or if it can be wrapped to do so. In the end what you should try is to have the pyramidal image stack for the main image as the `zarr_group` attribute. For some formats it's pretty trivial to write the ZarrStore interface that can then be loaded as a zarr group.  

I'm happy to give feedback on PRs, but can't guarantee prompt answers due to other projects having priority right now.

Also let me know if you need more information to get started.

Cheers,
Andreas ☺️

**Comment by ap--** [comment]
Just pinging for awareness: 

On the Bayer side @sdvillal @wolny -- is there someone to provide more immediate feedback for this these days?

Or does BigPicture have an interest in support for DICOM here @erikogabrielsson ?

**Comment by wolny** [comment]
From my side I'd also be happy to provide feedback on PRs.

Simple solution would be to add `wsidcom` or `pydicom` as dependency and provide an adapter which is compatible with `tiffslide`/`openslide` API, see: https://github.com/imi-bigpicture/wsidicom?tab=readme-ov-file#api-differences-between-wsidicom-and-openslide.

**Comment by sarthakpati** [comment]
@wolny: could you perhaps give an example of an existing adapter on tiffslide for @benmalef to build upon?

**Comment by jstriebel** [comment]
[highdicom](https://highdicom.readthedocs.io/en/latest/), [DCI](https://github.com/ImagingDataCommons)'s high-level Python Dicom library, supports arbitrary file-like interfaces, so it could also be considered as a dependency here: https://highdicom.readthedocs.io/en/latest/remote.html

It's based on top of pydicom, not sure which interface of the two is a better fit.
